nncf/onnx/graph/node_utils.py

```diff
@@ -171,10 +171,15 @@ def _get_activation_tensor_shape(
     :param target_point: Determines from input or ouput of a node take a shape info.
     :return: None, if there is no shape info, otherwise - tensor shape.
     """
+    # NOTE: Assumes that all output/input edges for the `node` with `output_port_id`/`input_port_id`
+    # equal to `target_point.port_id` should have the same `tensor_shape` value.
+
     if target_point.type == TargetType.PRE_LAYER_OPERATION:
-        shape = nncf_graph.get_input_edges(node)[target_point.port_id].tensor_shape
+        edges = [e for e in nncf_graph.get_input_edges(node) if e.input_port_id == target_point.port_id]
+        shape = edges[0].tensor_shape
```

Comment:
Are we able to wrap this and introduce the new method in the NNCFGraph to get a needed edge by port ID? Minor, just about better code re-usability.

Reply:
Good suggestion! I've added two methods `get_input_edge_by_port_id()` and ` get_output_edges_by_port_id()`.